datasets/ua/war_sanctions/crawler.py

```diff
@@ -0,0 +1,397 @@
+            h.apply_date(com_rep, "startDate", com_date)
+
+            context.emit(com_rep)
+
+    safety_manager = data.pop(
+        "Ship Safety Management Manager (IMO / Country / Date)", None
+    )
+    if safety_manager is not None:
+        safety_manager_parts = safety_manager.split(" / ")
+        if len(safety_manager_parts) == 3:
+            safety_name_imo, safety_country, safety_date = safety_manager_parts
+            if len(safety_name_imo.split(" (")) == 2:
+                safety_name, safety_imo = safety_name_imo.split(" (")
+            else:
+                overrides = lookup_override(context, safety_name_imo, "com_manager")
+                safety_name = overrides.get("name")
+                safety_imo = overrides.get("registrationCode")
+
+            safety_manager = context.make("LegalEntity")
+            safety_manager.id = context.make_id(safety_name, safety_country)
+            safety_manager.add("name", safety_name)
+            safety_manager.add("registrationNumber", safety_imo)
+            safety_manager.add("country", safety_country)
+            safety_manager.add("topics", "poi")
+            context.emit(safety_manager, target=True)
+
+            safety_rep = context.make("Representation")
+            safety_rep.id = context.make_id(
+                vessel.id, "safety managed by", safety_manager.id
+            )
+            safety_rep.add("client", vessel.id)
+            safety_rep.add("agent", safety_manager.id)
+            safety_rep.add("role", "Ship Safety Management Manager")
+            h.apply_date(safety_rep, "startDate", safety_date)
+
+            context.emit(safety_rep)
+
+    owner_info = data.pop("Shipowner (IMO / Country / Date)")
+    if owner_info != "":
+        owner_parts = owner_info.split(" / ")
+        if len(owner_parts) == 3:
+            owner_name_imo, owner_country, owner_date = owner_parts
+            if len(owner_name_imo.split(" (")) == 2:
+                owner_name, owner_imo = owner_name_imo.split(" (")
+            else:
+                overrides = lookup_override(context, owner_name_imo, "com_manager")
+                owner_name = overrides.get("name")
+                owner_imo = overrides.get("registrationCode")
+
+            owner = context.make("LegalEntity")
+            owner.id = context.make_id(owner_name, owner_country)
+            owner.add("name", owner_name)
+            owner.add("country", owner_country)
+            owner.add("registrationNumber", owner_imo)
+            owner.add("topics", "poi")
+            context.emit(owner, target=True)
+
+            ownership = context.make("Ownership")
+            ownership.id = context.make_id(vessel.id, "owned by", owner.id)
+            ownership.add("asset", vessel.id)
+            ownership.add("owner", owner.id)
+            ownership.add("ownershipType", "Owner")
+            h.apply_date(ownership, "startDate", owner_date)
+
+            context.emit(ownership)
```

Comment:
this looks ripe for factoring out to a dedicated function crawl_ship_owner(context, vessel, owner_info)

now that you have three functions that look almost exactly the same, maybe they can be refactored into one

```python
crawl_ship_relation(context, vessel, data: string, rel_schema, rel_role, from_prop, to_prop)
```

with three calls like

```python
crawl_ship_relation(
    context,
    vessel,
    data.pop("Shipowner (IMO / Country / Date)")
    "Owner",
    "asset",
    "owner",
)
```